src/LegacyStrings.jl

```diff
@@ -49,15 +49,15 @@ using Compat
     end
 
     if VERSION >= v"0.5.0-"
-        immutable ASCIIString <: DirectIndexString
+        struct ASCIIString <: DirectIndexString
```

Comment:
will any of the packages that use this still need to do so when they drop 0.5 support? it might be more worthwhile to do that upgrade downstream and make this package unnecessary, but I'm not sure about every use case